exec-agent/main.go

```diff
@@ -0,0 +1,132 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/eclipse/che/exec-agent/auth"
+	"github.com/eclipse/che/exec-agent/process"
+	"github.com/eclipse/che/exec-agent/rest"
+	"github.com/eclipse/che/exec-agent/rpc"
+	"github.com/eclipse/che/exec-agent/term"
+	"github.com/gorilla/mux"
+	"log"
+	"net/http"
+	"os"
+	"time"
+	"strings"
+)
+
+var (
+	AppHttpRoutes = []rest.RoutesGroup{
+		process.HttpRoutes,
+		rpc.HttpRoutes,
+		term.HttpRoutes,
+	}
+
+	AppOpRoutes = []rpc.RoutesGroup{
+		process.RpcRoutes,
+	}
+
+	serverAddress string
+	staticFlag    string
+	pathFlag      string
+)
+
+func init() {
+	// Server configuration
+	flag.StringVar(&serverAddress, "addr", ":9000", "IP:PORT or :PORT the address to start the server on")
+	flag.StringVar(&staticFlag, "static", "./static/", "path to static content")
+	flag.StringVar(&pathFlag, "path", "", "Path of the pty server. Go regexp syntax is suported.")
+
+	// Auth configuration
+	flag.BoolVar(&auth.Enabled, "enable-auth", false, "Whether authenticate on workspace master or not")
+	flag.StringVar(&auth.ApiEndpoint,
+		"auth-api-endpoint",
+		os.Getenv("CHE_API_ENDPOINT"),
+		"Auth api-endpoint, by default 'CHEAPI-ENDPOINT' environment variable is used for this")
+
+	// Process configuration
+	flag.IntVar(&process.CleanupPeriodInMinutes, "process-cleanup-period", 2, "How often processs cleanup will happen(in minutes)")
+	flag.IntVar(&process.CleanupThresholdInMinutes,
+		"process-lifetime",
+		60,
+		"How much time will dead and unused process live(in minutes), if -1 passed then processes won't be cleaned at all")
+	curDir, _ := os.Getwd()
+	curDir += string(os.PathSeparator) + "logs"
+	flag.StringVar(&process.LogsDir, "logs-dir", curDir, "Base directory for process logs")
+}
+
+func main() {
+	flag.Parse()
+
+	// print configuration
+	fmt.Println("Exec-agent configuration")
```

Comment:
I think code can be cleaner if you split it into methods `print summary, normalize base path, cleanup resources, setup routes, etc`